src/tye/Program.PurgeCommand.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.IO;
+using Microsoft.Tye.ConfigModel;
+using Microsoft.Tye.Hosting;
+
+namespace Microsoft.Tye
+{
+    static partial class Program
+    {
+        public static Command CreatePurgeCommand(string[] args)
```

Comment:
I'm not a fan of this command? Or maybe I'm just not a fan of the fact that we need to call into the runner to to a purge because the logic is there.

Reply:
I think it's the latter. The other question is whether we'd auto-purge if you do run again.

Reply:
@davidfowl It sounds reasonable that the entity which knows how to launch a certain type of replica would know how to kill replicas of that type.  
I don't think I like the idea of having a class like `ReplicaKiller` that switch/cases through the types of replicas.

The best alternative that I can think of is, you can have a virtual `Kill` method in the `ReplicaStatus` class (`ProcessStatus` and `DockerStatus` provide their own implementation)

Then, the `WriteReplicaEvent` in `ReplicaRegistry` would serialize the entire replica (as JSON?) together with its type.
`GetEvents` will return a list of `ReplicaStatus` (instead of list of `StoreEvent`) and the purge method will loop over all `ReplicaStatus`es and call `.Kill()`

How does it sound?

Reply:
Maybe a cleaner design is just to pass the registry to the docker runner and process runner. I recall me guiding you this direction but it seems like it might be much simpler now to delete the state when replicas are spun down in both the docker and process runner.

Reply:
@davidfowl  I could do it but how does it address your initial concern about the purge command or the fact that the runners contain the purge logic? Unless I'm misunderstanding something.

Reply:
Lets stick to auto purge on run and remove the purge command for now.

Reply:
@davidfowl Sure thing. I've removed the purge command but for now I've left the `PurgeAsyc` method in the host.
This is because the purging logic is in the runners and the host has access to create the runners, unless we change something structurally.
This can also be changed if I implement this
```
The best alternative that I can think of is, you can have a virtual Kill method in the ReplicaStatus class (ProcessStatus and DockerStatus provide their own implementation)
```

By the way, I've made this modification
```
while (!dockerInfo.StoppingTokenSource.Token.IsCancellationRequested)
{
    var logsRes = await ProcessUtil.RunAsync("docker", $"logs -f {containerId}",
        outputDataReceived: data => service.Logs.OnNext($"[{replica}]: {data}"),
        errorDataReceived: data => service.Logs.OnNext($"[{replica}]: {data}"),
        throwOnError: false,
        cancellationToken: dockerInfo.StoppingTokenSource.Token);

    if (logsRes.ExitCode != 0)
    {
        break;
    }
}
```
to https://github.com/dotnet/tye/pull/209 (added a break from the loop if the exit code is non zero)

Turns out that if you kill the container externally, it enters an infinite loop.
It caused my tests to fail but I also think that it's a good check anyway.
What do you think?

Reply:
- There should be more purge command

> Turns out that if you kill the container externally, it enters an infinite loop.
It caused my tests to fail but I also think that it's a good check anyway.
What do you think?

This seems fine if it normally doesn't infinitely loop when logs runs and ends.